File: src/Defense/anti_drone_mine.py
from mine import Mine
from power_ratings import EMP_POWER

from DefenseAdder.anti_drone_adder import AntiDroneAdder

class AntiDroneMine(Mine): # Not sure about the name, since it's kind of not a mine
    """ Represents the Anti-Drone Mine """
    adderClass = AntiDroneAdder
    powerRating = EMP_POWER
    
    def __init__(self):
        """ Initialize the Anti-Drone Mine """
        self.turrets = []
        Mine.__init__(self)
  
    def defuse(self, drone):
        """ Defuse the Mine """
        self.tryToDetonate(drone)

    def defuseCarefully(self, drone):
        """ Defuse the Mine carefully """
        self.tryToDetonate(drone)
  
    def hitByEMP(self, drone):
        """ Deactivat the Worm """
        self.defused = True
        
    def addTurret(self, turret):
        """ Add turret to the Anti Drone Mine """
        self.turrets.append(turret)

File: src/android/src/com/kaottessur/pymine/defense/AntiDroneMine.java
package com.kaottessur.pymine.defense;

import java.util.ArrayList;
import java.util.List;

import com.kaottessur.pymine.Power;
import com.kaottessur.pymine.level.Drone;

public class AntiDroneMine extends Mine {
	private List<AntiDroneTurret> turrets;
	
	public AntiDroneMine() {
		turrets = new ArrayList<AntiDroneTurret>();
	}
	
	@Override
	public int getPowerRating() {
		return Power.EMP_POWER*2;
	}
	
	@Override
	public void defuse(Drone drone) {
		deactivate();
	}

	@Override
	public void defuseCarefully(Drone drone) {
		deactivate();
	}

	@Override
	public void hitByEMP(Drone drone) {
		deactivate();
	}
	
	public void addTurret(AntiDroneTurret turret) {
		turrets.add(turret);
	}
	
	@Override
	public void deactivate() {
		super.deactivate();
		
		for (AntiDroneTurret turret : turrets) {
			turret.deactivate();
		}
	}
}
